ArduCopter/mode_auto.cpp

```diff
@@ -52,7 +52,9 @@ bool ModeAuto::init(bool ignore_checks)
 
 #if PRECISION_LANDING == ENABLED
         // initialise precland state machine
-        copter.precland_statemachine.init();
+        if (copter.precland.enabled()) {
+            copter.precland_statemachine.init();
```

Comment:
I think its better if the `precland_statemachine` just instantly returns if not enabled, you only need to check in one function then rather than each caller.

Reply:
Thanks for the review! I have fixed the issue